### DISCUSSION TITLE
No matter what I do, I can't generate a project

### DISCUSSION THREAD

**Discussion by Clos97** [ussserrr/stm32pio#121] [Q&A]
As you can see in the image below, I'm always getting a **FileNotFoundError** when I'm trying to generate a project.
I've used the example explaination shown in the documentation. I've generated code via CubeMX and when I'm trying to run the command `stm32pio new -b nucleo_l152re` I'm facing this issue.
![image](https://github.com/ussserrr/stm32pio/assets/148942966/4e7555af-acad-4cde-9a9c-40948202e4ec)

**Comment by ussserrr** [comment]
What's the content of the folder where you are running this command in?

**Reply by Clos97** [reply]
It's happening if I run it with just the ioc file in the directory or with a generated project including the sources.

**Reply by ussserrr** [reply]
Hmm, can you specify this file explicitly? Also, try with a verbose mode to collect more information.

**Comment by Clos97** [comment]
sure. 
these are the files that are inside of my project folder: 
![image](https://github.com/ussserrr/stm32pio/assets/148942966/53ce78d0-4b8b-4285-ba64-22818609de3d)
When I put the verbose flag into the command, the result looks like this:
![image](https://github.com/ussserrr/stm32pio/assets/148942966/ec15bc09-c3ff-4498-adb0-8e4a3d6b63f0)

**Reply by ussserrr** [reply]
Where is the CubeMX located? Does the default path (on screenshot) that stm32pio assumes when tries to invoke the CubeMX is sufficient for your installation? If not, you can set the correct one at the INI-config.

**Reply by Clos97** [reply]
The Cube MX path is the same as it is defined in the .ini file: 
"C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\STM32CubeMX.exe"

**Reply by Clos97** [reply]
Is there any update? Is the Path I'm using wrong or what's the issue?

**Reply by ussserrr** [reply]
The log saying there are some problems with invoking of CubeMX CLI but that's all. This issue is strange tbh, I don't think I have enough ability to solve it for your particular setup.

**Comment by Themuse061** [comment]
I've had the same error. After some digging I found that uninstalling STM32CubeMX and installing it again on version 6.3.0 (in default path i might add) did the trick. I think it could be because in later versions (you can check their release notes) they removed `Other Toolchains (GPDSC)` from their `Toolchain / IDE` selector in Project Manager. It's just my wild guess tho.
Also, i've used the terminal from ProjectIO. I followed [this ](https://www.youtube.com/watch?v=71cq0KUopGQ&t=418s)video as steps provided in cli example didn't work for me. 

![image_2025-03-09_195917508](https://github.com/user-attachments/assets/18e105ce-bf0f-41ba-8d7c-e592eb3104cb)
here is my platformio terminal log
